app/src/androidTest/java/org/mozilla/fenix/ui/NoNetworkAccessStartupTests.kt

```diff
@@ -0,0 +1,121 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.ui
+
+import androidx.core.net.toUri
+import androidx.test.espresso.Espresso
+import androidx.test.espresso.IdlingRegistry
+import org.junit.After
+import org.junit.Test
+import org.mozilla.fenix.R
+import org.mozilla.fenix.helpers.HomeActivityTestRule
+import org.mozilla.fenix.helpers.TestHelper.packageName
+import org.mozilla.fenix.helpers.TestHelper.setNetworkEnabled
+import org.mozilla.fenix.helpers.TestHelper.verifyUrl
+import org.mozilla.fenix.helpers.idlingresource.NetworkConnectionIdlingResource
+import org.mozilla.fenix.ui.robots.browserScreen
+import org.mozilla.fenix.ui.robots.homeScreen
+import org.mozilla.fenix.ui.robots.navigationToolbar
+
+/**
+ * Tests to verify some main UI flows with Network connection off
+ *
+ */
+
+class NoNetworkAccessStartupTests {
+    private val activityTestRule = HomeActivityTestRule()
+    private val networkDisconnectedIdlingResource = NetworkConnectionIdlingResource(false)
+    private val networkConnectedIdlingResource = NetworkConnectionIdlingResource(true)
+
+    @After
+    fun tearDown() {
+        IdlingRegistry.getInstance().unregister(networkDisconnectedIdlingResource)
+        // Restoring network connection and waiting to be back online
+        setNetworkEnabled(true)
+        IdlingRegistry.getInstance().register(networkConnectedIdlingResource)
+        Espresso.onIdle {
+            IdlingRegistry.getInstance().unregister(networkConnectedIdlingResource)
+        }
+
+        activityTestRule.finishActivity()
+    }
+
+    @Test
+    // Based on STR from https://github.com/mozilla-mobile/fenix/issues/16886
+    fun noNetworkConnectionStartupTest() {
+        setNetworkEnabled(false)
+        IdlingRegistry.getInstance().register(networkDisconnectedIdlingResource)
+        Espresso.onIdle {
+            IdlingRegistry.getInstance().unregister(networkDisconnectedIdlingResource)
+        }
+
+        activityTestRule.launchActivity(null)
```

Comment:
I think you need to do `HomeActivityTestRule(launchActivity = false)` to prevent the activity from already being launched by this point

Reply:
> I think it'd be fine to land even if we decided not to do them.

Oh actually, I think this one needs to be addressed for this test to work correctly. However, I trust you can implement the correct solution so no need to get a re-review from me before landing unless you want to.

Reply:
Ah, I added code to throw if there's no network on start up and the test runs as expected (i.e. it fails with that change). I think what might be happening is that we're launching the activity twice. So I guess it'd be an optimization to avoid launching the activity twice (if I understand what's happening correctly) but that this test is working correctly.

Reply:
done

Reply:
done